### PR TITLE
add sign spammer

### LINKED ISSUE #508: [Request] Sign Spammer Module
**Describe the solution you'd like**
A Module with 4 Text inputs in it's Settings, each representing a line of a sign. When the module is enabled, placing a sign will not open the text input screen but instead the lines in the settings will be placed on the sign and no GUI will open.

**Describe alternatives you've considered**
manually pasting the text onto each sign

### PR DESCRIPTION
**Pull request by LordMZTE**
closes #508

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/client/network/ClientPlayerEntity;openEditSignScreen(Lnet/minecraft/block/entity/SignBlockEntity;)V"
+            ),
+            cancellable = true,
+            locals = LocalCapture.CAPTURE_FAILSOFT
+    )
+    private void onSignEditorOpen(SignEditorOpenS2CPacket packet, CallbackInfo ci, BlockEntity blockEntity) {
+        // this variable is just here for some deduplication
+        SignSpammer spammer = SignSpammer.INSTANCE;
```
Comment (5HT2) [inline src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java:32]: I don't understand what this is supposed to mean. Do you mean avoiding typing SignSpammer.INSTANCE a bunch of times? If so, I think it doesn't matter if the Mixins aren't as pretty due to being in Java

Comment (LordMZTE) [reply src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java:32]: Yes, that's exactly what it means. I think it's still nice to have tho and makes the code look cleaner. Let's see what 086 thinks of it.

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/SignSpammer.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package me.zeroeightsix.kami.feature.module
+
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+
+@Module.Info(
+    name = "SignSpammer",
+    category = Module.Category.MISC,
+    description = "Automatically fills newly placed signs"
+)
+// implementation in MixinClientPlayNetworkHandler
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/feature/module/SignSpammer.kt]: Use a JavaDoc @see for this

File: src/main/resources/kami.mixins.json [hunk 3]
```diff
@@ -66,6 +66,7 @@
     "MixinRenderPlayer",
     "MixinRenderTickCounter",
     "MixinStriderEntity",
-    "MixinWorldRenderer"
+    "MixinWorldRenderer",
+    "MixinClientPlayNetworkHandler"
```
Comment (5HT2) [inline src/main/resources/kami.mixins.json]: Ctrl alt L inside intellij to alphabetize

**Review by 5HT2 — CHANGES_REQUESTED** [review]

File: src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java (hunk 1 above)
Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java:32]: > I think it doesn't matter if the Mixins aren't as pretty due to being in Java

Java can still be pretty. I'm happy with this variable.

**Review by zeroeightysix — APPROVED** [review]

File: src/main/kotlin/me/zeroeightsix/kami/feature/module/SignSpammer.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+@Module.Info(
+    name = "SignSpammer",
+    category = Module.Category.MISC,
+    description = "Automatically fills newly placed signs"
+)
+/**
+ * @see me.zeroeightsix.kami.mixin.client.MixinClientPlayNetworkHandler
+ */
+object SignSpammer : Module() {
```
Comment (5HT2) [inline src/main/kotlin/me/zeroeightsix/kami/feature/module/SignSpammer.kt]: this is more of a style thing but the javadoc should be above `@Module.Info`

**Review by 5HT2 — APPROVED** [review]
lgtm

**Comment by zeroeightysix** [comment]
If you want to: traditionally, this module works by duplicating the first sign you've written since enabling the module. If you were to add a button to the settings UI, something like `Copy from next sign` that replicates this behaviour (in a separate PR), that'd be cool!

**Comment by zeroeightysix** [comment]
Thanks!

**Pull request merged**